samples/drivers/adc/adc_sequence/boards/nrf54h20dk_nrf54h20_cpuapp.overlay

```diff
@@ -39,6 +39,16 @@
 		zephyr,oversampling = <8>;
 	};
 
+	channel@2 {
+		reg = <2>;
```

Comment:
You should also add `<&adc 2>` in `io-channels` above, otherwise this channel won't be actually used (only it's definition will be processed by `dtc`). Sorry that I haven't spotted that before.